--- gmt-5.1.0/cmake/modules/ConfigCMake.cmake.orig	(revision 12461)
+++ gmt-5.1.0/cmake/modules/ConfigCMake.cmake	(working copy)
@@ -177,23 +177,30 @@
 # (but later on when installing)
 set (CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
 
-# the RPATH to be used when installing
-set (CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/${GMT_LIBDIR}")
-
-# make executables relocatable on supported platforms
-if (UNIX AND NOT CYGWIN)
-	# find relative libdir from executable dir
-	file (RELATIVE_PATH _rpath /${GMT_BINDIR} /${GMT_LIBDIR})
-	# remove trailing /
-	string (REGEX REPLACE "/$" "" _rpath "${_rpath}")
+# set the RPATH to be used when installing
+if (GMT_INSTALL_RELOCATABLE)
+	# make executables relocatable on supported platforms (relative RPATH)
+	if (UNIX AND NOT CYGWIN)
+		# find relative libdir from executable dir
+		file (RELATIVE_PATH _rpath /${GMT_BINDIR} /${GMT_LIBDIR})
+		# remove trailing /
+		string (REGEX REPLACE "/$" "" _rpath "${_rpath}")
+		if (APPLE)
+			# relative RPATH on osx
+			set (CMAKE_INSTALL_NAME_DIR @executable_path/${_rpath})
+		else (APPLE)
+			# relative RPATH on Linux, Solaris, etc.
+			set (CMAKE_INSTALL_RPATH "\$ORIGIN/${_rpath}")
+		endif (APPLE)
+	endif (UNIX AND NOT CYGWIN)
+else (GMT_INSTALL_RELOCATABLE)
+	# set absolute RPATH
 	if (APPLE)
-		# relative RPATH on osx
-		set (CMAKE_INSTALL_NAME_DIR @executable_path/${_rpath})
+		set (CMAKE_INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/${GMT_LIBDIR}")
 	else (APPLE)
-		# relative RPATH on Linux, Solaris, etc.
-		set (CMAKE_INSTALL_RPATH "\$ORIGIN/${_rpath}")
+		set (CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/${GMT_LIBDIR}")
 	endif (APPLE)
-endif (UNIX AND NOT CYGWIN)
+endif (GMT_INSTALL_RELOCATABLE)
 
 # add the automatically determined parts of the RPATH
 # which point to directories outside the build tree to the install RPATH
